### PR TITLE
Adds Storybook as an app

### PR DESCRIPTION
**Pull request by conrad-vanl**
### What does this PR do, or why is it needed?

This creates a new storybook app as a separate package and loads stories from all of our frontend packages.

### What design trade-offs/decisions were made?

This introduces a second react-native app in core. The positive to this is that we could transition core-team development over to working 100% on storybook - building out features, components and screens directly within storybook. We could go as far as loading the entire `apolloschurchapp` prototype app as a story. This gives us the ability to have a 100% "playground" environment, while we can then work towards transitioning `apolloschurchapp` over to 100% boilerplate/starterkit project.

The trade-off is `apolloschurchapp` now has no built-in storybook, and so churches that create their own apollos project won't have storybook built in.

@vinnyjth - @burkeshartsis and I talked quite a bit about this and I'm not sure that we are 100% sold on the direction of this PR, nor are we certain what would be best. 

### How do I test this PR?

You can run the storybook app by `cd`'ing into the `ApollosStorybook` folder and running `yarn start`. It's just a typical react-native project.

## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,8 +6,6 @@
       "packages/*"
     ],
     "nohoist": [
-      "**/react-native",
```
Comment (conrad-vanl) [inline package.json]: I think we can add these back...but interesting if the app works without them might keep this removed?

File: packages/ApollosStorybook/native-storybook.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from '@storybook/react-native';
```
Comment (conrad-vanl) [inline packages/ApollosStorybook/native-storybook.js]: This allows all of our other `ui-` packages to import the same instance of the storybook dependency, instead of having to require their own (which doesn't work anyways)

File: package.json (hunk 1 above)
Comment (vinnyjth) [reply package.json]: Maybe. One thing I've noticed is that changes to `nohoist` don't always propagate to other's machines easily. Sometimes I'll push something that worked locally but find that `lerna` didn't apply the changes to my machine before I pushed, and so anyone who pulls and nukes will have a broken setup. 

tldr: We can know for sure after nuking :D

**Comment by vinnyjth** [comment]
We might have less conflicts if you create a new branch and cherry pick your commits, I think this branch is based on an out of date `onboarding-fixes` and so merging will be a mess.

**Comment by conrad-vanl** [comment]
> We might have less conflicts if you create a new branch and cherry pick your commits, I think this branch is based on an out of date onboarding-fixes and so merging will be a mess.

yeah i'm going to do this....
